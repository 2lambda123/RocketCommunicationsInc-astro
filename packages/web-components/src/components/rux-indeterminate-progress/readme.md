# rux-indeterminate-progress



<!-- Auto Generated Below -->


## Shadow Parts

| Part                  | Description                                                                                                                                    |
| --------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------- |
| `"animated-spinner"`  | The second div in the heirarchy which is responsible for the spinning conic-gradient and the gap between the spinner and the wrapper's border. |
| `"inner-circle"`      | The fourth div in the heirarchy which controls the inner-most circle.                                                                          |
| `"inner-spinner-gap"` | The third div in the heirarchy which controls the gap between the animated-spinner and the inner-circle's border.                              |
| `"wrapper"`           | The outermost div encasing rux-indeterminate-progress which is responsible for the overall size and the outermost border.                      |


## Dependencies

### Used by

 - [rux-progress](../rux-progress)

### Graph
```mermaid
graph TD;
  rux-progress --> rux-indeterminate-progress
  style rux-indeterminate-progress fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
