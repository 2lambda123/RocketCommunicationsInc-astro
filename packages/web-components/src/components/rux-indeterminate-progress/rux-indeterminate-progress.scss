:host {
    --size: 3.75rem;
    // These are used to determine nub-size
    --wrapper-height: calc(var(--size) + 4px);
    --second-div-height: calc(var(--size) * 0.9 - 6px);

    --nub-size: calc(
        ((var(--wrapper-height) - var(--second-div-height)) / 2) - 4px
    );

    // .5's here to pervent hard stops in mask
    --mask: radial-gradient(
        farthest-side,
        transparent calc((((100% - 4px) * 0.9) - 5.5px)),
        var(--color-background-interactive-default)
            calc((((100% - 4px) * 0.9) - 4.5px))
    );
    display: flex;
    align-items: center;
}
::slotted(*) {
    padding: 1.25rem;
}

.rux-indeterminate-wrapper {
    height: var(--size);
    width: var(--size);
    border-radius: 100%;
    display: flex;
    justify-content: center;
    align-items: center;
    border: 2px solid var(--color-border-interactive-default);
    background: var(--color-background-base-default);
    -webkit-mask: var(--mask);
    mask: var(--mask);
}

.rux-indeterminate-animated-spinner {
    position: relative;
    display: flex;
    justify-content: center;
    align-items: center;
    background: var(--indeterminate-gradient);
    /* h/w here increases distance of progress bar from top border*/
    height: calc(100% - 4px);
    width: calc(100% - 4px);
    border-radius: 100%;

    animation: spin 1.367s infinite linear;

    &::after {
        position: absolute;
        content: '';
        background: var(--indeterminate-nub-color);
        width: var(--nub-size);
        height: var(--nub-size);
        top: 0;
        margin: 0;
        padding: 0;
        border-radius: 100%;
    }
}

.rux-indeterminate-inner-spinner-gap {
    height: calc(90% - 6px);
    width: calc(90% - 6px);
    background: transparent;
    border-radius: 100%;
    border: 2px solid var(--color-background-base-default);
    display: flex;
    justify-content: center;
    align-items: center;
}
.rux-indeterminate-inner-circle {
    height: calc(100% - 4px);
    width: calc(100% - 4px);
    border: 2px solid var(--color-background-interactive-default);
    border-radius: 100%;
    background-color: transparent;
}

@keyframes spin {
    from {
        transform: rotate(0deg);
    }
    to {
        transform: rotate(360deg);
    }
}
