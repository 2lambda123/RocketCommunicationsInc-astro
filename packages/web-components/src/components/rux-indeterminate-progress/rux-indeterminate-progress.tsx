import { Component, Host, h } from '@stencil/core'

/**
 * @part wrapper - The outermost div encasing rux-indeterminate-progress which is responsible for the overall size and the outermost border.
 * @part animated-spinner - The second div in the heirarchy which is responsible for the spinning conic-gradient and the gap between the spinner and the wrapper's border.
 * @part inner-spinner-gap - The third div in the heirarchy which controls the gap between the animated-spinner and the inner-circle's border.
 * @part inner-cirlce - The fourth div in the heirarchy which controls the inner-most circle.
 */

@Component({
    tag: 'rux-indeterminate-progress',
    styleUrl: 'rux-indeterminate-progress.scss',
    shadow: true,
})
export class RuxIndeterminateProgress {
    render() {
        return (
            <Host>
                <div class="rux-indeterminate-wrapper" part="wrapper">
                    <div
                        class="rux-indeterminate-animated-spinner"
                        part="animated-spinner"
                    >
                        <div
                            class="rux-indeterminate-inner-spinner-gap"
                            part="inner-spinner-gap"
                        >
                            <div
                                class="rux-indeterminate-inner-circle"
                                part="inner-circle"
                            ></div>
                        </div>
                    </div>
                </div>
                <slot></slot>
            </Host>
        )
    }
}
