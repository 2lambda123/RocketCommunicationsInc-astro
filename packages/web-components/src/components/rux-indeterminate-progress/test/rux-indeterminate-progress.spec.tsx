import { newSpecPage } from '@stencil/core/testing'
import { RuxIndeterminateProgress } from '../rux-indeterminate-progress'

describe('rux-indeterminate-progress', () => {
    it('renders', async () => {
        const page = await newSpecPage({
            components: [RuxIndeterminateProgress],
            html: `<rux-indeterminate-progress></rux-indeterminate-progress>`,
        })
        expect(page.root).toMatchSnapshot()
    })
})
