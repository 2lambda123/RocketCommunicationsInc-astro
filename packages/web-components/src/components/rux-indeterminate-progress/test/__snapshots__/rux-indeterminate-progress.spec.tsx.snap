// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rux-indeterminate-progress renders 1`] = `
<rux-indeterminate-progress>
  <mock:shadow-root>
    <div class="rux-indeterminate-container" part="container">
      <div class="rux-indeterminate-animated-spinner" part="animated-spinner">
        <div class="rux-indeterminate-inner-spinner-gap" part="inner-spinner-gap">
          <div class="rux-indeterminate-inner-circle" part="inner-circle"></div>
        </div>
      </div>
    </div>
  </mock:shadow-root>
</rux-indeterminate-progress>
`;
